app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureActivity.kt

```diff
@@ -78,37 +88,52 @@ class EventCaptureActivity :
     var eventCaptureComponent: EventCaptureComponent? = null
     var programUid: String? = null
     var eventUid: String? = null
+    private var teiUid: String? = null
+    private var enrollmentUid: String? = null
     private val relationshipMapButton: LiveData<Boolean> = MutableLiveData(false)
     private var onEditionListener: OnEditionListener? = null
     private var adapter: EventCapturePagerAdapter? = null
+    private var eventViewPager: ViewPager2? = null
+    private var dashboardViewModel: DashboardViewModel? = null
     override fun onCreate(savedInstanceState: Bundle?) {
         eventUid = intent.getStringExtra(Constants.EVENT_UID)
-        eventCaptureComponent = this.app().userComponent()!!.plus(
-            EventCaptureModule(
-                this,
-                eventUid,
-            ),
-        )
-        eventCaptureComponent!!.inject(this)
+        teiUid = intent.getStringExtra(Constants.TEI_UID)
+        enrollmentUid = intent.getStringExtra(Constants.ENROLLMENT_UID)
+        programUid = intent.getStringExtra(Constants.PROGRAM_UID)
+        setUpEventCaptureComponent(eventUid)
         themeManager!!.setProgramTheme(intent.getStringExtra(Constants.PROGRAM_UID)!!)
         super.onCreate(savedInstanceState)
         binding = DataBindingUtil.setContentView(this, R.layout.activity_event_capture)
         binding.presenter = presenter
+        eventViewPager = when {
+            this.isLandscape() -> binding.eventViewLandPager
+            else -> binding.eventViewPager
+        }
         eventMode = intent.getSerializableExtra(Constants.EVENT_MODE) as EventMode?
         setUpViewPagerAdapter()
         setUpNavigationBar()
+        setUpEventCaptureFormLandscape(eventUid ?: "")
+        if (this.isLandscape() && areTeiUidAndEnrollmentUidNotNull()) {
+            val viewModelFactory = this.app().dashboardComponent()?.dashboardViewModelFactory()
+            viewModelFactory.let {
+                dashboardViewModel =
+                    ViewModelProvider(this, it!!)[DashboardViewModel::class.java]
+            }
+            supportFragmentManager.beginTransaction().replace(R.id.tei_column, newInstance(programUid, teiUid, enrollmentUid)).commit()
+            dashboardViewModel!!.updateSelectedEventUid(eventUid)
```

Comment:
what if we use the `?` operator and check that `viewModelFactory` is not null first. what do you think about this approach ?
`if (viewModelFactory != null) {
                    dashboardViewModel =
                            ViewModelProvider(this, viewModelFactory)[DashboardViewModel::class.java]
                supportFragmentManager.beginTransaction().replace(R.id.tei_column, newInstance(programUid, teiUid, enrollmentUid)).commit()
                dashboardViewModel?.updateSelectedEventUid(eventUid)
            }`

Reply:
seeing the code, I think that viewModelFactory is notNull if not, wouldn't be 
` viewModelFactory.let {...` it would be ` viewModelFactory?.let {...` so maybe there is no need to this viewModelFactory check, am I wrong?

on the other hand is there any case where dashboardViewModel can be null? just to understand the better solution

Reply:
Yes there are cases where viewModelFactory can be null, and that is because `val viewModelFactory = this.app().dashboardComponent()?.dashboardViewModelFactory()`  method might potentially return it as null. I think using `viewModelFactory?.let {...`  will be great.